test/hotspot/jtreg/compiler/c2/aarch64/TestStaticCallStub.java

```diff
@@ -43,52 +48,254 @@
+        }
+
+        protected final String opcode() {
+            return "movk";
+        }
+    }
+
+    private static class MOVZ extends Instruction {
+        public static final MOVZ INSTANCE = new MOVZ();
+
+        static {
+            Instruction.register(INSTANCE);
+        }
+
+        public static final MOVZ get() {
+            return INSTANCE;
+        }
+
+        protected final int mask() {
+            return 0b1111_1111_1000_0000_0000_0000_0000_0000;
+        }
+
+        protected final int value() {
+            return 0b1101_0010_1000_0000_0000_0000_0000_0000;
+        }
+
+        protected final String opcode() {
+            return "mov"; // this is not a typo
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+            // Main VM: fork VM with options
+            runVM(true);
+            runVM(false);
+            return;
+        }
+        if (args.length > 0) {
+            // We are in a forked VM. Just exit
+            System.out.println("Ok");
+        }
+    }
+
+    static List<String> extractOpcodeOrBytecodes(String line) {
+        int colonIndex = line.indexOf(':');
+        if (colonIndex != -1) {
+            line = line.substring(colonIndex + 1).trim();
+        }
 
-    static String extractOpcode(String line) {
-        line = line.trim();
         int semicolonIndex = line.indexOf(';');
         if (semicolonIndex != -1) {
             line = line.substring(0, semicolonIndex).trim();
         }
 
+        if (line.isBlank()) {
+            return Collections.emptyList();
+        }
+
         String[] words = line.split("\\s+");
-        if (words.length > 1) {
-            return words[1];
+        if (words.length > 0) {
+            String opcode = words[0];
+            // Does this look like a bytecode instead?
+            if (opcode.matches("^[0-9a-fA-F]{4}$")) {
+                List<String> retval = Arrays.stream(line.split("\\|"))
+                        .map(String::trim)
+                        .collect(Collectors.toList());
+                return retval;
+            }
+
+            return new ArrayList<>(Arrays.asList(opcode));
         }
 
-        return "";
+        return Collections.emptyList();
     }
 
-    static List<String> extractOpcodesN(ListIterator<String> itr, int n) {
-        List<String> extractedOpcodes = new ArrayList<>();
+    static List<String> extractCodesN(ListIterator<String> itr, int n) {
+        List<String> extracted = new ArrayList<>();
 
-        while (itr.hasNext() && extractedOpcodes.size() < n) {
-            String opcode = extractOpcode(itr.next());
-            if (!opcode.isEmpty()) {
-                extractedOpcodes.add(opcode);
-            }
+        while (itr.hasNext() && extracted.size() < n) {
+            int left = n - extracted.size();
+            extractOpcodeOrBytecodes(itr.next()).stream().limit(left).forEach(extracted::add);
```

Comment:
You can detect whether you have hex codes or disassembly.
See https://github.com/openjdk/jdk/blob/a86dd56de34f730b42593236f17118ef5ce4985a/test/hotspot/jtreg/compiler/onSpinWait/TestOnSpinWaitAArch64.java#L122

In TestOnSpinWaitAArch64 you can see how the checking code is organized not to depend on the instruction representation.

IMO you don't need the Instruction class hierarchy.
You need `nearStaticCallOpcodeSeq` and `farStaticCallOpcodeSeq` to be filled either with opcodes or hex codes. Of course their names need to be changed to something like `nearStaticCallInstSeq`. You will need to change `extractOpcodesN` and `extractOpcode` to `extractInstructionsN` and `extractInstruction`.